internal/pkg/service/common/httpserver/middleware/otel_apdex_test.go

```diff
@@ -0,0 +1,109 @@
+package middleware
+
+import (
+	"context"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/telemetry"
+)
+
+type testRequest struct {
+	DurationMs float64
+	StatusCode int
+}
+
+type testCase struct {
+	Name               string
+	Requests           []testRequest
+	ExpectedApdexT500  float64
+	ExpectedApdexT1000 float64
+}
+
+func TestApdex(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	cases := []testCase{
+		{
+			Name:               "server error, apdex=0",
+			Requests:           []testRequest{{DurationMs: 1, StatusCode: 500}, {DurationMs: 1, StatusCode: 503}},
+			ExpectedApdexT500:  0,
+			ExpectedApdexT1000: 0,
+		},
+		{
+			Name:               "client error, satisfied duration",
+			Requests:           []testRequest{{DurationMs: 1, StatusCode: 400}, {DurationMs: 1, StatusCode: 403}},
+			ExpectedApdexT500:  1,
+			ExpectedApdexT1000: 1,
+		},
+		{
+			Name:               "ok, satisfied duration",
+			Requests:           []testRequest{{DurationMs: 1, StatusCode: 200}, {DurationMs: 1, StatusCode: 201}},
+			ExpectedApdexT500:  1,
+			ExpectedApdexT1000: 1,
+		},
+		{
+			Name:               "ok, tolerated/satisfied duration",
+			Requests:           []testRequest{{DurationMs: 700, StatusCode: 200}, {DurationMs: 800, StatusCode: 201}},
+			ExpectedApdexT500:  0.5, // T=500ms < 700ms,800ms < 4T=2000ms --> tolerated range --> apdex=0.5
+			ExpectedApdexT1000: 1,   //           700ms,800ms <  T=1000ms --> satisfied range --> apdex=1
+		},
+		{
+			Name:               "ok, frustrated/tolerated duration",
+			Requests:           []testRequest{{DurationMs: 2400, StatusCode: 200}, {DurationMs: 2500, StatusCode: 201}},
+			ExpectedApdexT500:  0,   // 2400ms,2500ms > 4T=2000ms            --> frustrated range --> apdex=0
+			ExpectedApdexT1000: 0.5, // T=1000ms < 2400ms,2500ms < 4T=4000ms --> tolerated range  --> apdex=0.5
+		},
+		{
+			Name:               "ok, frustrated duration",
+			Requests:           []testRequest{{DurationMs: 10000, StatusCode: 200}, {DurationMs: 11000, StatusCode: 201}},
+			ExpectedApdexT500:  0,
+			ExpectedApdexT1000: 0,
+		},
+		{
+			Name: "mix",
+			Requests: []testRequest{
+				{DurationMs: 10000, StatusCode: 200},
+				{DurationMs: 11000, StatusCode: 201},
+				{DurationMs: 700, StatusCode: 200},
+				{DurationMs: 800, StatusCode: 201},
+				{DurationMs: 1, StatusCode: 400},
+				{DurationMs: 1, StatusCode: 403},
+				{DurationMs: 1, StatusCode: 500},
+				{DurationMs: 1, StatusCode: 503},
+			},
+			ExpectedApdexT500:  0.375, // (satisfied:2 + tolerated:2/2)/all:8
```

Comment:
Chvílu sem přemýšlal proč sů satisfied enom dva když sů 4 pod threshold než sem sa podíval na code :)

Reply:
Ostatne su 5xx, je to aj v komentare uplne hore `Satisfied request:  no server error AND duration < T`, mam to dat este niekde inde?

Reply:
asi nemusíš enom mě na první pohled netrklo že ty dva nejsů satisfied kvůli status code